yamlscanner/scanner.py

```diff
@@ -0,0 +1,141 @@
+import asyncio
+import contextlib
+import discord
+import yaml
+
+from redbot.core import commands
+from redbot.core.utils.menus import start_adding_reactions
+from redbot.core.utils.chat_formatting import pagify, box, humanize_list
+from redbot.core.utils.predicates import ReactionPredicate, MessagePredicate
+
+
+class YamlScanner(commands.Cog):
+    """An easy and quick tool to validate yaml."""
+
+    __author__ = ["Kreusada"]
+    __version__ = "1.0.0"
+
+
+    def __init__(self, bot):
+        self.bot = bot
+
+
+    def format_help_for_context(self, ctx: commands.Context) -> str:
+        context = super().format_help_for_context(ctx)
+        authors = humanize_list(self.__author__)
+        return f"{context}\n\nAuthor: {authors}\nVersion: {self.__version__}"
+
+
+    async def red_delete_data_for_user(self, **kwargs):
+        """Nothing to delete"""
+        return
+
+
+    def cog_unload(self):
+        with contextlib.suppress(Exception):
+            self.bot.remove_dev_env_value("raffle")
+
+
+    async def initialize(self) -> None:
+        if 719988449867989142 in self.bot.owner_ids:
+            with contextlib.suppress(Exception):
+                self.bot.add_dev_env_value("raffle", lambda x: self)
+
+
+    @commands.command(usage="[file]")
+    async def yamlscan(self, ctx: commands.Context):
+        """Scan yaml to see if its correct.
+
+        Your next message will be used to as the yaml to scan.
+        You can also upload a YAML file.
+        """
+
+        def cleanup_code(content) -> str:
+            # From redbot.core.dev_commands, thanks will :P
+            if content.startswith("```") and content.endswith("```"):
+                return "\n".join(content.split("\n")[1:-1])
+            return content.strip("` \n")
+
+        def tick(text) -> str:
+            return "{} {}".format("\N{BALLOT BOX WITH CHECK}\N{VARIATION SELECTOR-16}", text)
+        
+        def cross(text) -> str:
+            return "{} {}".format("\N{CROSS MARK}", text)
+
+        if ctx.message.attachments[0]:
+            # attachments will take priority
+            file = ctx.message.attachments[0]
+            filename = file.filename
+            if not filename.split('.')[-1] in ("yaml", "yml", "mir"):
```

Comment:
As this is the only place that `filename` appears it doesn't make too much sense to make a separate variable for it (Optional)